### PR TITLE
Update SharedTree2's factory type

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

The type "SharedTree" is already generally the `type` written into attributes blobs for documents using `@fluid-experimental/tree`. Re-using this string means that it will be harder to distinguish between `@fluid-experimental/tree` and `@fluid-experimental/tree2` (and may require changes to the process of) while loading the summary.

This will be a headache for migration scenarios, so we should make them distinct while we don't support back-compat.

## Breaking Changes

This breaks any persisted documents which use `@fluid-experimental/tree2` unless the application author modifies their container runtime's data store factory registry to contain an entry pointing "SharedTree" at tree2.

### CODE REVIEW

File: experimental/dds/tree2/src/shared-tree/sharedTree.ts [hunk 1]
```diff
@@ -243,7 +243,7 @@ export const defaultSharedTreeOptions: Required<SharedTreeOptions> = {
  * @alpha
  */
 export class SharedTreeFactory implements IChannelFactory {
-	public type: string = "SharedTree";
+	public type: string = "https://graph.microsoft.com/types/tree";
```
Comment (Abe27342) [inline experimental/dds/tree2/src/shared-tree/sharedTree.ts:246]: I'm not very opinionated on what we pick here. This is the style used for shared-string / matrix / pact-map, but I think the choices here come from aspirations to do more dynamic code loading than the types of scenarios we really aim to support these days.

Comment (CraigMacomber) [reply experimental/dds/tree2/src/shared-tree/sharedTree.ts:246]: see also typed-tree, the other factory that wraps shared tree as that was by far the easiest way to get to our MVP API. I think that one matters more than this one, and also has the same issue.

**Comment by CraigMacomber** [comment]
I think ideally PRs which break all existing documents should include a changeset to they get an entry in the release notes. Also clarity on what error we get would be nice, and help ensure we know what errors for different things look lke and that they are comprehensable to people who get them.

File: experimental/dds/tree2/src/shared-tree/sharedTree.ts (hunk 1 above)
Comment (Abe27342) [reply experimental/dds/tree2/src/shared-tree/sharedTree.ts:246]: Where is that defined? Not finding references to it.

Comment (Abe27342) [reply experimental/dds/tree2/src/shared-tree/sharedTree.ts:247]: nvm, needed to merge main. Fixed in latest.

**Review by tyler-cai-microsoft — APPROVED** [review]
LGTM :shipit:

**Pull request merged**